### PR TITLE
Paging: do not show all the posts at once, use pages instead

### PR DESCRIPTION
**Pull request by nicordev**

### CODE REVIEW

File: src/Application.php [hunk 1]
```diff
@@ -59,7 +67,7 @@ public function run()
                 $method = new ReflectionMethod($route->controller, $route->method);
 
             } catch (ReflectionException $e) {
-                throw new AppException('The method ' . $route->method . ' was not found in ' . $route->controller);
+                throw new AppException('The method ' . $route->method . ' was not found in ' . $route->controller, 0, $e);
```
Comment (olix21) [inline src/Application.php]: In this case an HttpException could be thrown, with a 404 code

File: src/Controller/BlogController.php [hunk 2]
```diff
@@ -664,7 +716,7 @@ private function addNewTagsFromTagList(array $tagIds, array $tagNames)
                 try {
                     $this->tagManager->add(new Tag(['name' => $tagNames[$i]]));
                 } catch (Exception $e) {
-                    throw new AppException('Impossible to add the tag ' . $tagNames[$i]);
+                    throw new AppException('Impossible to add the tag ' . $tagNames[$i], 0, $e);
```
Comment (olix21) [inline src/Controller/BlogController.php]: code 500 (internal server error)

File: src/Controller/BlogController.php [hunk 3]
```diff
@@ -700,7 +752,7 @@ private function updateTag(Tag $tagToUpdate, array $tagIds, array $tagNames)
                 try {
                     $this->tagManager->edit($updatedTag);
                 } catch (Exception $e) {
-                    throw new AppException('Impossible to edit the tag ' . print_r($tagData, true));
+                    throw new AppException('Impossible to edit the tag ' . print_r($tagData, true), 0, $e);
```
Comment (olix21) [inline src/Controller/BlogController.php]: code 500 (internal server error)

**Comment by nicordev** [comment]
I added the changes.

**Pull request merged**